lib/services/collections_service.dart

```diff
@@ -308,18 +313,47 @@ class CollectionsService {
   }
 
   void _cacheCollectionAttributes(Collection collection) {
+    _collectionIDToCollections[collection.id] = collection;
+    final updatedCollection = getCollectionWithDecryptedName(collection);
     if (collection.attributes.encryptedPath != null) {
-      _localCollections[decryptCollectionPath(collection)] = collection;
+      _localCollections[decryptCollectionPath(collection)] = updatedCollection;
     }
-    _collectionIDToCollections[collection.id] = collection;
+    _collectionIDToCollections[collection.id] = updatedCollection;
   }
 
   String decryptCollectionPath(Collection collection) {
+    final key = collection.attributes.version == 1
+        ? getCollectionKey(collection.id)
+        : _config.getKey();
     return utf8.decode(CryptoUtil.decryptSync(
         Sodium.base642bin(collection.attributes.encryptedPath),
-        _config.getKey(),
+        key,
         Sodium.base642bin(collection.attributes.pathDecryptionNonce)));
   }
+
+  Collection getCollectionWithDecryptedName(Collection collection) {
+    var name;
+    if (collection.encryptedName != null && collection.encryptedName != "") {
```

Comment:
Instead of `collection.encryptedName != ""`, you can do a `collection.encryptedName.isNotEmpty`.